File: File Upload Validator/cpp/validator.cpp
#include <iostream>
#include <fstream>
#include <vector>

int main(int argc, char *argv[]) {
    if (argc < 2) return 1;
    std::ifstream file(argv[1], std::ios::binary);
    if (!file.is_open()) { std::cout << "ERROR" << std::endl; return 1; }

    char buffer[4];
    file.read(buffer, 4);
    if (file.gcount() < 3) { std::cout << "INVALID" << std::endl; return 0; }

    if ((unsigned char)buffer[0] == 0xFF && (unsigned char)buffer[1] == 0xD8 && (unsigned char)buffer[2] == 0xFF) {
        std::cout << "VALID_JPG" << std::endl; return 0;
    }
    if (file.gcount() >= 4 && buffer[0] == 0x25 && buffer[1] == 0x50 && buffer[2] == 0x44 && buffer[3] == 0x46) {
        std::cout << "VALID_PDF" << std::endl; return 0;
    }
    std::cout << "INVALID" << std::endl;
    return 0;
}

File: File Upload Validator/python/validator.py
import sys
import os

class FileValidator:
    SIGNATURE_JPG = b'\xFF\xD8\xFF'
    SIGNATURE_PDF = b'\x25\x50\x44\x46'

    @staticmethod
    def validate_file(file_path):
        try:
            with open(file_path, 'rb') as f:
                header = f.read(4)
                if header.startswith(FileValidator.SIGNATURE_JPG): return "VALID_JPG"
                if header.startswith(FileValidator.SIGNATURE_PDF): return "VALID_PDF"
                return "INVALID"
        except: return "ERROR"

if __name__ == "__main__":
    if len(sys.argv) < 2:
        print("Usage: validator.py <file>")
        sys.exit(1)
    print(FileValidator.validate_file(sys.argv[1]))
